python/ngen_cal/src/ngen/cal/optimizers/__init__.py

```diff
@@ -0,0 +1 @@
+from .._optimizers.grey_wolf import GreyWolfOptimizer
```

Comment:
This refactor adds a layer of indirection to safeguard the `optimizers`
public api. The `_optimizers` subpackage provides a place for _both_
stable and unstable / experimental optimizers to live. Once an
optimizer's api and features are fleshed out, it can then be imported by
the `optimizers` subpackage in `__init__.py` effectively moving it into
the public api.

Reply:
Gotcha.  I was thrown off a bit by having in my head that the class was still experimental, since the description for the PR notes it not being exposed yet.

Reply:
Yeah, it is a little confusing. I also thought we should not expose it to the public api until we have it "hooked up," but @hellkite500 thought otherwise.